MatrixMultiplication/MatrixMultiplication/MatrixMultiplication/Matrix.cs

```diff
@@ -0,0 +1,241 @@
+﻿namespace MatrixMultiplication;
+
+public class Matrix
+{
+    /// <summary>
+    /// Number of rows in the matrix
+    /// </summary>
+    public int NumberOfRows { get; private set; }
+
+    /// <summary>
+    /// Number of cols in the matrix
+    /// </summary>
+    public int NumberOfCols { get; private set; }
+
+    private readonly int[,] data;
+    static readonly Random random = new();
+
+    /// <summary>
+    /// Indexer
+    /// </summary>
+    /// <param name="i">row index</param>
+    /// <param name="j">column index</param>
+    /// <returns></returns>
+    public int this[int i, int j]
+    {
+        get => data[i, j];
+        set => data[i, j] = value;
+    }
+
+    /// <summary>
+    /// Two-dimensional array for matrix initialization
+    /// </summary>
+    /// <param name="data"></param>
+    public Matrix(int[,] data)
+    {
+        this.data = data;
+        this.NumberOfRows = data.GetLength(0);
+        this.NumberOfCols = data.GetLength(1);
+    }
+
+    /// <summary>
+    /// Function for matrix multiplication
+    /// </summary>
+    /// <param name="matrix">First matrix</param>
+    /// <param name="strategy">Second matrix</param>
+    /// <returns>Result of multiplication</returns>
+    public Matrix Multiply(Matrix matrix, IStrategy strategy)
+    {
+        return strategy.Multiply(this, matrix);
+    }
```

Comment:
```suggestion
    public Matrix Multiply(Matrix matrix, IStrategy strategy)
        => strategy.Multiply(this, matrix);
```
Это даже по сути --- просто делегация.

Reply:
Не поправлено